test/e2e/retry_test.go

```diff
@@ -48,6 +49,35 @@ spec:
 		})
 }
 
+func (s *RetryTestSuite) TestRetryBackoff() {
+	s.Given().
+		Workflow(`
+metadata:
+  generateName: test-backoff-strategy-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      retryStrategy:
+        limit: 10
+        backoff:
+          duration: 10s
+          maxDuration: 50s
+      container:
+          name: main
+          image: 'argoproj/argosay:v2'
+          args: [ exit, "1" ]
+`).
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(time.Minute).
+		Then().
+		ExpectWorkflow(func(t *testing.T, _ *metav1.ObjectMeta, status *wfv1.WorkflowStatus) {
+			assert.Equal(t, wfv1.WorkflowPhase("Failed"), status.Phase)
+			assert.Equal(t, "Max duration limit exceeded", status.Message)
```

Comment:
I think you need to validate the number of retries since the issue is "string" `limit` does not get interpret correctly

Reply:
In this case, we could check the length of child nodes at the end of the run which should be 5 based on the backoff time.